galasa-ui/src/components/test-runs/LogTab.tsx

```diff
@@ -311,6 +401,75 @@ export default function LogTab({ logs, initialLine }: LogTabProps) {
     return result;
   };
 
+  // Effect to clear selection state when the user deselects text anywhere.
+  useEffect(() => {
+    const handleSelectionChange = () => {
+      const selection = window.getSelection();
+      if (selection && selection.isCollapsed) {
+        setSelectedRange(null);
+      }
+    };
+
+    document.addEventListener("selectionchange", handleSelectionChange);
+
+    return () => {
+      document.removeEventListener("selectionchange", handleSelectionChange);
+    };
+  }, []);
+
+  // Effect to listen to the browser hash changes and updates the state that runs only on mount
+  useEffect(() => {
+    const handleHashChange = () => {
+      setCurrentHash(window.location.hash);
+    };
+
+    window.addEventListener('hashchange', handleHashChange);
+    return () => {
+      window.removeEventListener('hashchange', handleHashChange);
```

Comment:
Minor - we could store the event listeners in constants so we don't have to keep track of their occurrences if they ever change rather than string literals.